### PR TITLE
RSDK-9218: Change all of the artifacts to v4.

### PR DESCRIPTION
**Pull request by dgottlieb**

### CODE REVIEW

File: .github/workflows/code-coverage-comment.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
     if: ${{ github.event.workflow_run.event == 'pull_request_target' && github.event.workflow_run.conclusion == 'success' }}
     steps:
       - name: Download Code Coverage
-        uses: dawidd6/action-download-artifact@v2
+        uses: actions/action-download-artifact@v4
```
Comment (abe-winter) [inline .github/workflows/code-coverage-comment.yml:15]: p sure this doesn't exist
https://github.com/actions/action-download-artifact

**Review by abe-winter — APPROVED** [review]
lgtm other than the code coverage / motion bench changes which will likely fail when run

File: .github/workflows/code-coverage-comment.yml (hunk 1 above)
Comment (abe-winter) [reply .github/workflows/code-coverage-comment.yml:15]: it looks like the official action added the 'other workflows' feature
https://github.com/actions/download-artifact?tab=readme-ov-file#download-artifacts-from-other-workflow-runs-or-repositories

guessing it's a drop-in change if you switch run_id -> run-id

**Comment by dgottlieb** [comment]
Test failure was https://viam.atlassian.net/browse/RSDK-8779

Plowing through the unsuccessful checks

**Pull request merged**